### PR TITLE
Cassandra support for chat history

### PR DESCRIPTION
**Pull request by jj701**
# Cassandra support for chat history

### Description

- Store chat messages in cassandra

### Dependency

- cassandra-driver - Python Module

## Before submitting

- Added Integration Test

## Who can review?

@hwchase17
@agola11

### CODE REVIEW

File: langchain/memory/chat_message_histories/cassandra.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        self,
+        contact_points: List[str],
+        session_id: str,
+        port: int = DEFAULT_PORT,
+        username: str = DEFAULT_USERNAME,
+        password: str = DEFAULT_PASSWORD,
+        keyspace_name: str = DEFAULT_KEYSPACE_NAME,
+        table_name: str = DEFAULT_TABLE_NAME,
+    ):
+        from cassandra.cluster import Cluster, PlainTextAuthProvider
```
Comment (leo-gan) [inline langchain/memory/chat_message_histories/cassandra.py:46]: Maybe import it in a try block, like:
```
        try:
            from cassandra.cluster import Cluster, PlainTextAuthProvider
        except ImportError:
            raise ValueError(
                "Could not import cassandra-driver python package. "
                "Please install it with `pip install cassandra-driver`."
            )
```

Comment (jj701) [reply langchain/memory/chat_message_histories/cassandra.py]: Added this

File: langchain/memory/chat_message_histories/cassandra.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        try:
+            self.cluster: Cluster = Cluster(
+                contact_points,
+                port=port,
+                auth_provider=PlainTextAuthProvider(
+                    username=self.username, password=self.password
+                ),
+            )
+            self.session = self.cluster.connect()
+        except Exception as error:
```
[the remark this answers is not in the document]
Comment (jj701) [reply langchain/memory/chat_message_histories/cassandra.py]: I have made the changes let me know

**Review by dev2049 — COMMENTED** [review]
thanks @jj701! high level looks good, will take a closer look in a bit. could we add a simple example notebook to `docs/modules/memory/examples/`?

**Comment by jj701** [comment]
> thanks @jj701! high level looks good, will take a closer look in a bit. could we add a simple example notebook to `docs/modules/memory/examples/`?

Added the Jupyter Notebook Example

**Pull request merged**